cursor autofocuses to answer text box on load

rckim77 opened the pull request.

rckim77 committed:
cursor autofocuses to answer text box on load

elliottyates commented:
Because this is actually a modal that is displayed and redisplayed, the second time you open a quiz me modal the focus doesn't go to the textfield. You may have to find a way to do this on the modal load using Javascript instead.

elliottyates commented:
Also if you have bandwidth on your personal heroku account, if you can deploy this as a shareable branch so we don't have to download and run it locally that would be great.

rckim77 committed:
autofocus textarea now written in js

rckim77 commented:
here are the updates on my heroku app: https://evening-scrubland-7568.herokuapp.com/

elliottyates commented on app/views/qsets/_modal_question.erb:
```diff
@@ -87,5 +87,9 @@
     })
+
+    $('#question_display_Modal').on('shown.bs.modal', function (event) {
+      $('textarea').focus();
```
Note this will work because there are no other `textarea` elements anywhere. If there were, this would not work as expected, because this selector would return a list of the `textarea` elements on the page. So this works, but it would be best to be more specific -- given there's only one quiz-me modal on the page we could attach a unique `id` attribute to that textarea element and select it here instead of this more general selector. It would future-proof us.

rckim77 committed:
future-proofed answer box selection by adding an id to textarea

rckim77 committed:
undo last commit

rckim77 committed:
future-proofed answer box by adding id for textarea

rckim77 commented:
I made a commit to origin master by accident so I just made a new commit to undo it and then made the proper commit to cursor_autofocus. I added an id to `<textarea>` called "answer-box" and in the `<script>` tag I changed the selector to #answer-box.

elliottyates commented:
:+1:
